crates/bevy_pbr/src/ssao/mod.rs

```diff
@@ -0,0 +1,958 @@
+                    BindGroupLayoutEntry {
+                        binding: 1,
+                        visibility: ShaderStages::COMPUTE,
+                        ty: BindingType::Buffer {
+                            ty: BufferBindingType::Uniform,
+                            has_dynamic_offset: true,
+                            min_binding_size: Some(ViewUniform::min_size()),
+                        },
+                        count: None,
+                    },
+                ],
+            });
+
+        let mip_texture_entry = BindGroupLayoutEntry {
+            binding: 1,
+            visibility: ShaderStages::COMPUTE,
+            ty: BindingType::StorageTexture {
+                access: StorageTextureAccess::WriteOnly,
+                format: TextureFormat::R16Float,
+                view_dimension: TextureViewDimension::D2,
+            },
+            count: None,
+        };
+        let preprocess_depth_bind_group_layout =
+            render_device.create_bind_group_layout(&BindGroupLayoutDescriptor {
+                label: Some("ssao_preprocess_depth_bind_group_layout"),
+                entries: &[
+                    BindGroupLayoutEntry {
+                        binding: 0,
+                        visibility: ShaderStages::COMPUTE,
+                        ty: BindingType::Texture {
+                            sample_type: TextureSampleType::Depth,
+                            view_dimension: TextureViewDimension::D2,
+                            multisampled: false,
+                        },
+                        count: None,
+                    },
+                    mip_texture_entry,
+                    BindGroupLayoutEntry {
+                        binding: 2,
+                        ..mip_texture_entry
+                    },
+                    BindGroupLayoutEntry {
+                        binding: 3,
+                        ..mip_texture_entry
+                    },
+                    BindGroupLayoutEntry {
+                        binding: 4,
+                        ..mip_texture_entry
+                    },
+                    BindGroupLayoutEntry {
+                        binding: 5,
+                        ..mip_texture_entry
+                    },
+                ],
+            });
+
+        let gtao_bind_group_layout =
+            render_device.create_bind_group_layout(&BindGroupLayoutDescriptor {
+                label: Some("ssao_gtao_bind_group_layout"),
+                entries: &[
+                    BindGroupLayoutEntry {
+                        binding: 0,
+                        visibility: ShaderStages::COMPUTE,
+                        ty: BindingType::Texture {
+                            sample_type: TextureSampleType::Float { filterable: false },
+                            view_dimension: TextureViewDimension::D2,
+                            multisampled: false,
+                        },
+                        count: None,
+                    },
```

Comment:
Rather than do anything in this PR, I'd rather tackle the issue in bevy's rendering infrastructure itself. I have some ideas on how to simplify this kind of thing.